services/ui-src/src/components/menus/Sidebar.tsx

```diff
@@ -1,190 +1,250 @@
+          </Box>
+          <Box sx={sx.topBox}>
+            <Heading sx={sx.title}>MCPAR Report Submission Form</Heading>
+          </Box>
+          {LinkItems.map((section) => (
+            <NavSection
+              key={section.name}
+              section={section}
+              level={1}
+              basePath="/mcpar"
+            />
+          ))}
+        </Box>
+      )}
+    </>
   );
 };
 
-const CloseButton = () => {
+interface NavSectionProps {
+  key: string;
+  section: LinkItemProps;
+  level: number;
+  basePath: string;
+}
+
+const NavSection = ({ section, level, basePath }: NavSectionProps) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const { name, path, children } = section;
+  const itemPath = `${basePath}${path}`;
   return (
-    <Flex align="center" paddingY="0.5rem" marginLeft="1.25rem" sx={sx.temp}>
-      <ArrowIcon title="closeNavBarButton" direction="left" />
-    </Flex>
+    <>
+      <React.Fragment key={itemPath}>
+        {children ? (
+          <Box as="button" onClick={() => setIsOpen(!isOpen)}>
+            <NavItem name={name} level={level} hasChildren={!!children} />
+          </Box>
+        ) : (
+          <>
+            <Link as={RouterLink} to={itemPath}>
+              <NavItem name={name} level={level} hasChildren={!!children} />
+            </Link>
+          </>
+        )}
+        {!!children && (
+          <Collapse in={isOpen}>
+            {children.map((section) => (
+              <NavSection
+                key={section.name}
+                section={section}
+                level={level + 1}
+                basePath={itemPath}
+              />
+            ))}
+          </Collapse>
+        )}
+      </React.Fragment>
+    </>
   );
 };
 
-interface NavItemProps extends FlexProps {
-  path: string;
+interface NavItemProps {
   name: string;
-  hasSubItems: boolean;
-  itemLevel?: string;
-  children?: any;
-  onClick?: any;
+  level: number;
+  hasChildren: boolean;
 }
-const NavItem = ({
-  path,
-  name,
-  hasSubItems,
-  itemLevel,
-  children,
-  onClick,
-  ...rest
-}: NavItemProps) => {
-  const linkPath = window.location.origin + "/mcpar/" + path;
-  return (
-    // <Link href={linkPath} textColor="palette.gray_darkest">
-    <Flex sx={sx.navItemFlex} {...rest} className={itemLevel}>
-      <CheckCircleIcon viewBox="10 10 200 200" />
-      <Text>{name}</Text>
-      <Flex direction="column">{children}</Flex>
-      {hasSubItems && (
-        <Button onClick={onClick}>
-          <ArrowIcon title="openNavItemsArrow" direction="down" />
-        </Button>
-      )}
-    </Flex>
-    // </Link>
-  );
-};
+
+const NavItem = ({ name, level, hasChildren }: NavItemProps) => (
+  <Flex sx={sx.navItemFlex}>
+    <CheckCircleIcon />
+    <Text sx={sx.navItemTitle} className={`level-${level}`}>
+      {name}
+    </Text>
+    {!!hasChildren && <ArrowIcon title="openNavItemsArrow" direction="down" />}
+  </Flex>
+);
```

Comment:
And here's our friend `NavItem`. Every item renders one of these, wrapped in either a button or link as you saw above. I moved the icons a bit and absolute positioned them, and put the text in the middle with a class for styling.